python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py

```diff
@@ -84,35 +53,20 @@ def build_component_defs(
         f"{Path(components_root).parent.name}.{Path(components_root).name}"
     )
 
-    return load_defs(defs_root=defs_root, resources=resources, component_types=component_types)
+    return load_defs(defs_root=defs_root, resources=resources)
 
 
 # Public method so optional Nones are fine
 @suppress_dagster_warnings
 def load_defs(
-    defs_root: ModuleType,
-    resources: Optional[Mapping[str, object]] = None,
-    component_types: Optional[dict[ComponentKey, type[Component]]] = None,
+    defs_root: ModuleType, resources: Optional[Mapping[str, object]] = None
 ) -> "Definitions":
     """Constructs a Definitions object, loading all Dagster defs in the given module.
 
     Args:
         defs_root (Path): The path to the defs root, typically `package.defs`.
         resources (Optional[Mapping[str, object]]): A mapping of resource keys to resources
             to apply to the definitions.
-        component_types (Optional[dict[ComponentKey, type[Component]]]): A mapping of
-            component keys to component types.
     """
-    from dagster._core.definitions.definitions_class import Definitions
-
-    component_types = component_types or discover_entry_point_component_types()
-    components_root_dir = get_path_from_module(defs_root)
-
-    all_defs: list[Definitions] = []
     module_cache = DefinitionsModuleCache(resources=resources or {})
-    for component_path in [item for item in components_root_dir.iterdir() if item.is_dir()]:
-        defs = module_cache.load_defs(
-            module=importlib.import_module(f"{defs_root.__name__}.{component_path.name}"),
-        )
-        all_defs.append(defs)
-    return Definitions.merge(*all_defs)
+    return module_cache.load_defs(defs_root)
```

Comment:
Beyond being cleaner, this area was actually the source of a few bugs, including the fact that putting a bare python file at the root of your `defs/` directory would not get picked up by the loading system (note that the previous code iterated over directories in the top-level defs folder, meaning any files there would be ignored)

Reply:
However, this does mean that the DefinitionsModuleCache no longer does anything all that useful, as its caching granularity is too wide.

This caching logic needs to be reworked upstack so that we can actually cache incremental bits, which I think would make the most sense to do on the `DefinitionsModule` class itself.

Just leaving this as is for now because marginally worse perf is not really a concern at the moment and I don't want to make this diff bigger than it already is cc @benpankow